packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -378,6 +396,51 @@ export abstract class AbstractMessageManager<
   rejectMessage(messageId: string) {
     this.setMessageStatus(messageId, 'rejected');
   }
+
+  /**
+   * Creates a promise which will resolve or reject when the message process is finished.
+   *
+   * @param messageParamsWithId - The params for the personal_sign call to be made after the message is approved.
+   * @param messageName - The name of the message
+   * @param hasErrorState - Whether the message has an error state.
+   * @returns Promise resolving to the raw data of the signature request.
+   */
+  async waitForFinishStatus(
+    messageParamsWithId: AbstractMessageParamsMetamask,
+    messageName: string,
+    hasErrorState = false,
```

Comment:
As you've moved the error state to the `AbstractMessage`, we don't need this argument anymore as it will simply never be set for any messages except typed messages.

Reply:
Done